tests/repoproviders_test.py

```diff
@@ -0,0 +1,58 @@
+from unittest import TestCase
+
+import pytest
+
+from binderhub.repoproviders import GitHubRepoProvider
+
+
+# General string processing
+@pytest.mark.parametrize(
+    'raw_text, suffix, clean_text', [
+        ("foo.git", ".git", "foo"),
+        ("foo.bar", ".git", "foo.bar"),
+        ("foo.bar", ".bar", "foo")
+    ]
+)
+def test_string_strip(raw_text, suffix, clean_text):
+    assert GitHubRepoProvider.strip_suffix(raw_text, suffix) == clean_text
+
+
+# user/repo/reference
+@pytest.mark.parametrize(
+    'spec, raw_user, raw_repo, raw_ref', [
+        ("user/repo/master", "user", "repo", "master"),
+        ("user/repo/hotfix/squash-bug", "user", "repo", "hotfix/squash-bug"),
+        ("user/repo/feature/save_world", "user", "repo", "feature/save_world")
+    ]
+)
+def test_spec_processing(spec, raw_user, raw_repo, raw_ref):
+    user, repo, unresolved_ref = GitHubRepoProvider.process_spec(spec)
+    assert raw_user == user
+    assert raw_repo == repo
+    assert raw_ref == unresolved_ref
+
+
+class TestSpecErrorHandling(TestCase):
+
+    def test_too_short_spec(self):
+        spec = "nothing_to_split"
+        with self.assertRaisesRegexp(ValueError, "Spec is not of the form"):
```

Comment:
Not sure if this is a technical or just stylistic choice: I try to stick as much as possible with the tools from pytest, so `with pytest.raises(...)` https://docs.pytest.org/en/latest/assert.html#assertions-about-expected-exceptions instead of `unittest` things.

Reply:
The main reason for my usage of `unittest.TestCase` was because I'm fond of the `assert.RaisesRegexp` method, from my quick reading of `pytest` it only allows for strict-equality-matching the error message.

I haven't used `pytest`'s assertions styles much, but I can convert if you'd like since I defer to you guys (the core project maintainers) for setting testing style standards :)